### DISCUSSION TITLE
Equivalent of `git add --patch` split hunks option?

### DISCUSSION THREAD

**Discussion by fuzzbomb** [magit/magit#4529] [Support] [answered]
I've been using Magit for a few days, and like it a lot so far!

There's one thing I miss from using Git at the command line: splitting the current hunk during interactive patching.

Magit presents separate hunks in the UI, and they can be staged individually by pressing `s`. So far, so good. That's equivalent to using `git add --patch` and answering `y` to "stage this hunk?".

Sometimes Git will identify a hunk, which I want to split up so I can stage part of it. In this situation during a `git add --patch` session, I would answer `s` to "split the current hunk into smaller hunks".  Does Magit have an equivalent to this?

I realize that I can make visual region selections within Magit (using Emacs own mark/point region, or Evil's visual-line selections) to stage parts of a hunk. However there's a bit more effort (physical and cognitive) because I have to mark the start and end lines myself. The split-hunk feature of interactive staging breaks things down for me, so it's just about saying yes/no at a more granular level. There's no need for me to put my mark/point in particular places.

This isn't a deal-breaker for me at all, just something I'm curious about.

**Comment by tarsius** [comment, marked as answer]
Make the hunks smaller by pressing `-`.
